frontend/pulse3d/src/components/interactiveAnalysis/WaveformGraph.js

```diff
@@ -552,31 +571,148 @@ export default function WaveformGraph({
       })
       .on("drag", function (d) {
         const id = d3.select(this).attr("id");
+        const yId = id.includes("1") ? "y1" : "y2";
 
-        // ensure you can't move a line outside of window bounds
-        const yPosition =
-          id == "peakLine" ? Math.max(d.y, y(yMax + yRange)) : Math.min(d.y, y(yMin - yRange));
+        const yPosition = id.includes("valley")
+          ? Math.max(d.y, y(yMax + yRange))
+          : Math.min(d.y, y(yMin - yRange));
+        //set new y for marker
+        d3.select(this).attr("cy", yPosition);
 
-        // assign new x values
-        d3.select(this).attr("y1", yPosition).attr("y2", yPosition);
+        //set new y for line
+        if (id.includes("valley")) {
+          d3.select("#valleyLine").attr(yId, yPosition);
+        } else {
+          d3.select("#peakLine").attr(yId, yPosition);
+        }
       })
-      .on("end", function () {
-        const id = d3.select(this).attr("id");
-        const yPosition = d3.select(this).attr("y1");
+      .on("end", function (d) {
+        // descrease stroke width when unselected and dropped
+        d3.select(this).attr("stroke-width", 2);
 
-        d3.select(this).attr("y1", yPosition).attr("y2", yPosition);
+        const id = d3.select(this).attr("id");
 
-        const pvCopy = peakValleyWindows;
-        if (id === "peakLine") {
-          pvCopy[currentWell].minPeaks = y.invert(yPosition);
-          setPeakValleyWindows({ ...pvCopy });
+        //need to invert to make calculation compatible with dataToGrpah
+        const y1 = y.invert(
+          id.includes("valley") ? d3.select("#valleyLine").attr("y1") : d3.select("#peakLine").attr("y1")
+        );
+        const y2 = y.invert(
+          id.includes("valley") ? d3.select("#valleyLine").attr("y2") : d3.select("#peakLine").attr("y2")
+        );
+
+        const x1 = x.invert(d3.select("#peakLineY1Marker").attr("cx"));
+        const x2 = x.invert(d3.select("#peakLineY2Marker").attr("cx"));
+
+        const slope = (y2 - y1) / (x2 - x1);
+        const yintersept = y2 - slope * x2;
+        // set variables used in marker deletion calculation
+        if (id.includes("peak")) {
```

Comment:
Both branches of this if/else are very similar, there is probably a way to reduce the amount of duplicate code written here

Reply:
The first place I would start (as mentioned in one of my other comments) is making a function for this same 3 lines of code that is repeated over and over here:
```
const updateArray = (updateFn, originalArray, newVal) => {
  let newArr = [...originalArray];
  newArr[wellIndex] = newVal; // could also parametrize wellIndex
  updateFn(newArr);
};
```
Now we can swap in the function:
```
if (id.includes("peak")) {
  updateArray(setPeakY1, peakY1, y1);
  updateArray(setPeakY2, peakY2, y2);
  updateArray(setPeakYIntercept, peakYIntercept, yIntercept);
  updateArray(setPeakSlope, peakSlope, slope);
} else {
  updateArray(setValleyY1, valleyY1, y1);
  updateArray(setValleyY2, valleyY2, y2);
  updateArray(setValleyYIntercept, valleyYIntercept, yIntercept);
  updateArray(setValleySlope, valleySlope, slope);
}
```
At this point another pattern becomes apparent, `updateArray` is being called over and over. There are few different ways reduce down to one line of code actually calling `updateArray`, here's just one:
```
const updateArgs = id.includes("peak")
      ? [
          [setPeakY1, peakY1, y1],
          [setPeakY2, peakY2, y2],
          [setPeakYIntercept, peakYIntercept, yIntercept],
          [setPeakSlope, peakSlope, slope],
        ]
      : [
          [setValleyY1, valleyY1, y1],
          [setValleyY2, valleyY2, y2],
          [setValleyYIntercept, valleyYIntercept, yIntercept],
          [setValleySlope, valleySlope, slope],
        ];
    for (const args of updateArgs) updateArray(...args);
```
You could argue readability was reduced by this change however, so I think it's fine to stick with just the changes I mentioned prior to this